podcast-player/src/components/episode/EpisodeList.jsx

```diff
@@ -0,0 +1,85 @@
+import React, { useState } from "react";
+import { useEffect } from "react";
+import { getEpisodes } from "../../utils/api-utils";
+import { ReactComponent as PlayButton } from "../../assets/SVG/play2.svg";
+
+import EpisodeModal from "./EpisodeModal";
+
+function EpisodeList({
+  show,
+  selectedEpisodePlaying,
+  setSelectedEpisodePlaying,
+}) {
+  const [episodes, setEpisodes] = useState(null);
+  const [selectedShow, setSelectedShow] = useState({});
+  const [selectedEpisode, setSelectedEpisode] = useState();
+  const [firstTenEpisodes, setFirstTenEpisodes] = useState([]);
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    let isCancelled = false;
+
+    if (show && !isCancelled) {
+      setIsLoading(true);
+      getEpisodes(show.feed)
+        .then((data) => {
+          setSelectedShow({ ...data.channel });
+          setEpisodes([...data.channel.episodes]);
+          // temp limiting data
+          setFirstTenEpisodes([...data.channel.episodes].slice(0, 9));
```

Comment:
Unnecessary shallow copying, but beyond that -

No need to have this all as separate state hooks. You're not copying from props, but there's a similar issue described here: https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html

You can just save the `data` and use that.